File: frontend/src/services/BookingService.js
import api from './ApiService';

// 🔹 Create a new booking
const createBooking = async (bookingData) => {
  const res = await api.post('/api/bookings', bookingData);
  return res.data;
};

// 🔹 Get bookings for current logged-in user
const getMyBookings = async () => {
  const res = await api.get('/api/bookings/user/me');
  return res.data;
};

// 🔹 Get bookings for a specific listing (admin/owner)
const getBookingsByListingId = async (listingId) => {
  const res = await api.get(`/api/bookings/listing/${listingId}`);
  return res.data;
};

// 🔹 Cancel a booking by ID
const cancelBooking = async (bookingId) => {
  const res = await api.put(`/api/bookings/${bookingId}/cancel`);
  return res.data;
};

// 🔹 Get details of a specific booking
const getBookingById = async (bookingId) => {
  const res = await api.get(`/api/bookings/${bookingId}`);
  return res.data;
};

// 🔹 Get bookings filtered by status (admin/owner)
const getBookingsByStatus = async (status) => {
  const res = await api.get(`/api/bookings/status/${status}`);
  return res.data;
};

const getBookingsByOWner = async() =>{
  const res = await api.get('/api/bookings/owner');
  return res.data;
}

const bookingAction = async (bookingId, action) => {
  const res = await api.post(
    `/api/bookings/listing/booking/action?bookingId=${bookingId}&action=${action}`
  );
  return res.data;
};

// ✅ Export all functions
const bookingService = {
  createBooking,
  getMyBookings,
  getBookingsByListingId,
  cancelBooking,
  getBookingById,
  getBookingsByStatus,
  getBookingsByOWner,
  bookingAction
};

export default bookingService;

File: backend/src/main/java/com/mit/StayNest/Services/BookingService.java
package com.mit.StayNest.Services;

import com.mit.StayNest.Entity.Booking;
import com.mit.StayNest.Entity.Listing;
import com.mit.StayNest.Entity.Owner;
import com.mit.StayNest.Entity.User;

import java.util.List;
import java.util.Optional;

public interface BookingService {

    Booking createBooking(Booking booking);

    List<Booking> getBookingsByTenant(User tenant);

    List<Booking> getBookingsByListing(Listing listing);

    Optional<Booking> getBookingById(Long id);

    List<Booking> getBookingsByStatus(String status);

    Booking cancelBooking(Long id);
    

    void updateBookingStatus(Long bookingId, String action, Owner owner);


    double getRentForBooking(Long id , double month);

    List<Booking> getBookingsByOwner(Owner owner);
}
